pybsn/switch.py

```diff
@@ -0,0 +1,86 @@
+class Switch(object): 
+    """Switch based API operations
+
+    """
+    def __init__(self, client):
+        self.client = client
+
+        # Default configurable values
+        self.name = None
+        self.dpid = None
+        self.fabric_role = None
+        self.leaf_group = None
+
+    @staticmethod
+    def get_switches(client, filter=None):
+
+        if filter:
```

Comment:
So `filter` is a dict, but this function only uses one item from it?

Maybe it would be better to have `get_switches` always return everything, and `get_switch_by_name` and `get_switch_by_dpid` return a single switch.